pkg/command/glauth.go

```diff
@@ -0,0 +1,41 @@
+package command
+
+import (
+	"github.com/micro/cli/v2"
+	"github.com/owncloud/ocis-glauth/pkg/command"
+	svcconfig "github.com/owncloud/ocis-glauth/pkg/config"
+	"github.com/owncloud/ocis-glauth/pkg/flagset"
+	"github.com/owncloud/ocis/pkg/config"
+	"github.com/owncloud/ocis/pkg/register"
+)
+
+// GLAuthCommand is the entrypoint for the glauth command.
+func GLAuthCommand(cfg *config.Config) *cli.Command {
+	return &cli.Command{
+		Name:     "glauth",
+		Usage:    "Start glauth server",
+		Category: "Extensions",
+		Flags:    flagset.ServerWithConfig(cfg.GLAuth),
+		Action: func(c *cli.Context) error {
+			scfg := configureGLAuth(cfg)
+
+			return cli.HandleAction(
+				command.Server(scfg).Action,
+				c,
+			)
+		},
+	}
+}
+
+func configureGLAuth(cfg *config.Config) *svcconfig.Config {
+	cfg.GLAuth.Log.Level = cfg.Log.Level
+	cfg.GLAuth.Log.Pretty = cfg.Log.Pretty
+	cfg.GLAuth.Log.Color = cfg.Log.Color
+	cfg.GLAuth.Tracing.Enabled = false
+	cfg.GLAuth.Ldap.Address = "localhost:9125"
```

Comment:
Should this be hard coded?

Reply:
We are hardcoding several services to localhost when running them via ocis. It is a question what ocis is intended to provide: an easy developer deployment or a tool to start a production deployment? The former would mean locking things to localhost may be the better approach. The latter would lead to a more flexible but secure by default deployment.

IMO letting all services listen to localhost might be ok for a development setup ... and should be the default for all extensions. Then ocis does not need to hardcode the host and address in the subcommands like above. And all services started via ocis can be configured va env vars, eg. to let the proxy listen for connections on all ip adresses.

@refs @micbar @PVince81 @C0rby @IljaN @felixboehm opinions? localhost by default? would that work for every service?

Reply:
IMO localhost as default, could be configured via ENV variables

Reply:
tracked in https://github.com/owncloud/ocis/issues/146